app/model/commands/Command.scala

```diff
@@ -1,9 +1,17 @@
 package model.commands
 
 import data.UnpackedDataStores
+import com.gu.pandomainauth.model.{User => PandaUser}
 
 trait Command extends UnpackedDataStores {
   type T
 
   def process(): T
+
+  def getUsername(user: PandaUser): String = {
+    user.email match {
+      case "" => user.firstName
+      case _ => user.email
```

Comment:
HMAC authed users have a `""` email.

I did [try](https://github.com/guardian/panda-hmac/pull/2) to fix this a while ago but failed and this was a quick win.